tests/test_representation.py

```diff
@@ -21,131 +23,215 @@ def load_tests(loader, tests, ignore):
     return tests
 
 
-class TestRepresentation(PandasTestCase):
-    """
-    Count.
-    """
-
-    def test_count_single_document(self):
-        s = pd.Series("a b c c")
-        s = preprocessing.tokenize(s)
-        s_true = pd.Series([[1, 1, 2]])
-        self.assertEqual(representation.count(s), s_true)
-
-    def test_count_multiple_documents(self):
-        s = pd.Series(["doc_one", "doc_two"])
-        s = preprocessing.tokenize(s)
-        s_true = pd.Series([[1, 0], [0, 1]])
-        self.assertEqual(representation.count(s), s_true)
-
-    def test_count_not_lowercase(self):
-        s = pd.Series(["one ONE"])
-        s = preprocessing.tokenize(s)
-        s_true = pd.Series([[1, 1]])
-        self.assertEqual(representation.count(s), s_true)
+"""
+Test functions in representation module in a
+parameterized way.
+"""
 
-    def test_count_punctuation_are_kept(self):
-        s = pd.Series(["one !"])
-        s = preprocessing.tokenize(s)
-        s_true = pd.Series([[1, 1]])
-        self.assertEqual(representation.count(s), s_true)
 
-    def test_count_not_tokenized_yet(self):
-        s = pd.Series("a b c c")
-        s_true = pd.Series([[1, 1, 2]])
+# Define valid inputs / outputs / indexes for different functions.
+s_not_tokenized = pd.Series(["This is not tokenized!"])
+s_tokenized = pd.Series([["Test", "Test", "TEST", "!"], ["Test", "?", ".", "."]])
+s_tokenized_with_noncontinuous_index = pd.Series(
+    [["Test", "Test", "TEST", "!"], ["Test", "?", ".", "."]], index=[5, 7]
+)
+
+s_tokenized_output_index = pd.MultiIndex.from_tuples(
+    [(0, "!"), (0, "TEST"), (0, "Test"), (1, "."), (1, "?"), (1, "Test")],
+    names=["document", "word"],
+)
+
+s_tokenized_output_noncontinuous_index = pd.MultiIndex.from_tuples(
+    [(5, "!"), (5, "TEST"), (5, "Test"), (7, "."), (7, "?"), (7, "Test")],
+    names=["document", "word"],
+)
+
+s_tokenized_output_min_df_index = pd.MultiIndex.from_tuples(
+    [(0, "Test"), (1, "Test")], names=["document", "word"],
+)
+
+
+test_cases_vectorization = [
+    # format: [function_name, function, correct output for tokenized input above, dtype of output]
+    ["count", representation.count, [1, 1, 2, 2, 1, 1], "int"],
+    [
+        "term_frequency",
+        representation.term_frequency,
+        [0.125, 0.125, 0.250, 0.250, 0.125, 0.125],
+        "float",
+    ],
+    [
+        "tfidf",
+        representation.tfidf,
+        [1.405465, 1.405465, 2.000000, 2.810930, 1.405465, 1.000000],
```

Comment:
It would be great if we avoid hardcoding these values there and use a function instead.

That way we can really check that the function is using the exact formula we mention in the docstring:

```
 The *inverse document frequency (idf)* measures how _important_ or _characteristic_
    a term is among the whole corpus (i.e. among all documents).
    Thus, idf(term t) = log((1 + number of documents) / (1 + number of documents where t is present)) + 1.
    Finally, tf-idf(document d, term t) = tf(d, t) * idf(t)
```

Reply:
Will do